### PR TITLE
Add flag to skip the dart analyzer error check before a hot reload

### PR DESCRIPTION
**Pull request by DaveShuckerow**
Currently, if the analysis server detects any errors in the project, it will not perform a hot reload.
This can cause hot reload to stop working needlessly when, eg, there is an analysis error in a test file.
 
Take two on #3181.

### CODE REVIEW

File: src/io/flutter/FlutterBundle.properties [hunk 1]
```diff
@@ -228,3 +228,4 @@ settings.open.inspector.on.launch=Open Flutter Inspector view on app launch
 settings.hot.reload.on.save=Perform hot reload on save
 settings.disable.tracking.widget.creation=Disable tracking widget creation locations
 settings.enable.bazel.test.runner=Enable new Bazel test runner
+settings.hot.reload.with.error=Perform hot reload even if there are analysis errors in the project
```
Comment (devoncarew) [inline src/io/flutter/FlutterBundle.properties]: Perhaps, `Perform hot reload even if there are analysis errors`?

File: src/io/flutter/run/FlutterReloadManager.java [hunk 2]
```diff
@@ -190,14 +190,16 @@ private void handleSaveAllNotification(@Nullable Editor editor) {
         return;
       }
 
-      if (hasErrors(app.getProject(), app.getModule(), editor.getDocument())) {
+      // If the analysis server detects any errors in the project, it will not perform a hot reload.
+      // This can cause hot reload to stop working needlessly when, eg, there is an analysis error in a test file.
+      // The reloadWithError option in settings is a workaround.
+      if (hasErrors(app.getProject(), app.getModule(), editor.getDocument()) && !mySettings.isReloadWithError()) {
```
Comment (devoncarew) [inline src/io/flutter/run/FlutterReloadManager.java]: Is there an issue tracking a better fix (in flutter_tools)? If so, we could in-line the reference here.

File: src/io/flutter/sdk/FlutterSettingsConfigurable.form [hunk 3]
```diff
@@ -196,13 +196,22 @@
...
               <toolTipText value="Allows the Flutter Inspector to track source locations where widgets are created. This setting is required for advanced Flutter Inspector functionality and performance tools."/>
             </properties>
           </component>
+          <component id="ff6b2" class="javax.swing.JCheckBox" binding="myHotReloadIgnoreErrorCheckBox">
+            <constraints>
+              <grid row="1" column="0" row-span="1" col-span="2" vsize-policy="0" hsize-policy="3" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
+            </constraints>
+            <properties>
+              <text resource-bundle="io/flutter/FlutterBundle" key="settings.hot.reload.with.error"/>
+              <toolTipText value="Hot reload changes into running Flutter apps even with analysis errors."/>
```
Comment (devoncarew) [inline src/io/flutter/sdk/FlutterSettingsConfigurable.form]: `Hot reload changes into running Flutter apps even if there are analysis errors`?

Comment (DaveShuckerow) [reply src/io/flutter/sdk/FlutterSettingsConfigurable.form]: Done.

File: src/io/flutter/run/FlutterReloadManager.java (hunk 2 above)
Comment (DaveShuckerow) [reply src/io/flutter/run/FlutterReloadManager.java]: Done.

File: src/io/flutter/FlutterBundle.properties (hunk 1 above)
Comment (DaveShuckerow) [reply src/io/flutter/FlutterBundle.properties]: Done.

**Pull request merged**